### PR TITLE
feat: Synchronous decoupled token store

### PR DESCRIPTION
**Pull request by PhilippeWeidmann**

### CODE REVIEW

**Comment by PhilippeWeidmann** [comment]
Duplication issues will be resolved when merging #830

File: MailCore/Cache/TokenStore.swift [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+import InfomaniakCore
+import InfomaniakDI
+
+public class TokenStore {
+    public enum TokenStoreFetchLocation {
+        case cache
```
Comment (valentinperignon) [inline MailCore/Cache/TokenStore.swift:25]: Do we prefer 
```swift
case cache, keychain
```
or
```swift
case cache
case keychain
```
I noticed that I often write only one `case` but have no preference

File: Mail/Views/Alerts/LogoutConfirmationView.swift [hunk 2]
```diff
@@ -43,12 +43,19 @@ struct LogoutConfirmationView: View {
...
         }
-        AccountManager.instance.removeTokenAndAccount(token: account.token)
-        if let nextAccount = AccountManager.instance.accounts.first {
-            AccountManager.instance.switchAccount(newAccount: nextAccount)
-        }
-        AccountManager.instance.saveAccounts()
+    }
+}
+
+extension ApiToken: Identifiable {
```
Comment (valentinperignon) [inline Mail/Views/Alerts/LogoutConfirmationView.swift:56]: Why ApiToken needs to be Identifiable here?

File: Mail/Views/Switch User/AccountView.swift [hunk 3]
```diff
@@ -27,13 +27,15 @@ import SwiftUI
 
 class AccountViewDelegate: DeleteAccountDelegate {
     @MainActor func didCompleteDeleteAccount() {
-        guard let account = AccountManager.instance.getCurrentAccount() else { return }
-        AccountManager.instance.removeTokenAndAccount(token: account.token)
-        if let nextAccount = AccountManager.instance.accounts.first {
-            AccountManager.instance.switchAccount(newAccount: nextAccount)
-            IKSnackBar.showSnackBar(message: "Account deleted")
+        Task {
```
Comment (valentinperignon) [inline Mail/Views/Switch User/AccountView.swift:30]: Can you explain me why we need to create a task

**Review by valentinperignon — COMMENTED** [review]
👍🏻

File: Mail/Views/Switch User/AccountView.swift (hunk 3 above)
Comment (PhilippeWeidmann) [reply Mail/Views/Switch User/AccountView.swift:30]: We don't, at first TokenStore was an Actor and we had to await every call but I couldn't do this unless a lot of refactor was done so I stayed with our thread safe (but blocking) collections

File: Mail/Views/Alerts/LogoutConfirmationView.swift (hunk 2 above)
Comment (PhilippeWeidmann) [reply Mail/Views/Alerts/LogoutConfirmationView.swift]: It needs to be identifiable to be presented as a an item in the sheet before. I moved this extension in the view that this presenting the logout confirmation

File: MailCore/Cache/TokenStore.swift (hunk 1 above)
Comment (PhilippeWeidmann) [reply MailCore/Cache/TokenStore.swift:25]: I do not have a preference either but I would tend to say that multiple line cases improve readability and are easier to refactor if we add more cases in the future.

**Review by valentinperignon — APPROVED** [review]

**Pull request merged**